beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBatchFetcher.java

```diff
@@ -178,53 +193,136 @@ private boolean batchIsComplete() {
         .orElse(false);
   }
 
-  private SafeFuture<Boolean> requestBlocksByRange() {
+  private SafeFuture<Boolean> requestBlocksAndBlobSidecarsByRange() {
     final RequestParameters requestParams = calculateRequestParams();
     if (requestParams.getCount().equals(UInt64.ZERO)) {
       // Nothing left to request
       return SafeFuture.completedFuture(false);
     }
 
+    final UInt64 endSlot = requestParams.getEndSlot();
+
+    final RequestManager requestManager =
+        new RequestManager(
+            lastBlockRoot,
+            getLatestReceivedBlock(),
+            blocksToImport::addLast,
+            this::processBlobSidecar);
+
     LOG.trace(
         "Request {} blocks from {} to {}",
         requestParams.getCount(),
         requestParams.getStartSlot(),
-        requestParams.getEndSlot());
+        endSlot);
+
+    final SafeFuture<Void> blocksRequest =
+        peer.requestBlocksByRange(
+            requestParams.getStartSlot(), requestParams.getCount(), requestManager::processBlock);
+
+    final SafeFuture<Void> blobSidecarsRequest;
+    if (blobsSidecarManager.isAvailabilityRequiredAtSlot(endSlot)) {
+      LOG.trace(
+          "Request {} blob sidecars from {} to {}",
+          requestParams.getCount(),
+          requestParams.getStartSlot(),
+          endSlot);
+      blobSidecarsRequest =
+          peer.requestBlobSidecarsByRange(
+              requestParams.getStartSlot(),
+              requestParams.getCount(),
+              requestManager::processBlobSidecar);
+    } else {
+      blobSidecarsRequest = SafeFuture.COMPLETE;
+    }
 
-    final RequestManager requestManager =
-        new RequestManager(lastBlockRoot, getLatestReceivedBlock(), blocksToImport::addLast);
+    return SafeFuture.allOfFailFast(blocksRequest, blobSidecarsRequest)
+        .thenApply(__ -> shouldRetryBlocksAndBlobSidecarsByRangeRequest());
+  }
 
-    return peer.requestBlocksByRange(
-            requestParams.getStartSlot(), requestParams.getCount(), requestManager::processBlock)
-        .thenApply(__ -> shouldRetryBlocksByRangeRequest());
+  private void processBlobSidecar(final BlobSidecar blobSidecar) {
+    blobSidecarsByBlockRootToImport
+        .computeIfAbsent(blobSidecar.getBlockRoot(), __ -> new ArrayList<>())
+        .add(blobSidecar);
   }
 
-  private boolean shouldRetryBlocksByRangeRequest() {
+  private boolean shouldRetryBlocksAndBlobSidecarsByRangeRequest() {
     return !batchIsComplete() && requestCount.incrementAndGet() < maxRequests;
   }
 
-  private SafeFuture<Void> requestBlockByHash() {
-    LOG.trace("Request next historical block directly by hash {}", lastBlockRoot);
+  private SafeFuture<Void> requestBlockByRoot() {
```

Comment:
We where discussing with @zilm13 that jumping gaps with by root works only for recent blocks, peers are not expected to serve the byRoot rpc for old blocks. Which means:
1- we should remove it and use only by range.
2- do another fallback to byRange (if by root fails) and from that moment use only that

this could be a dedicated PR :)

Reply:
Cool, so use byRange but only for one slot in this case? It will make things slightly easier actually as well since can just request blob sidecars rather than calculate number of commitments and then create BlobIdentifier list.